README.md

```diff
@@ -482,6 +483,33 @@ class { '::sensu::backend':
 }
 ```
 
+### Composite Names for Namespaces
+
+All resources that support having a `namespace` also support a composite name to define the namespace.
+
+For example, the `sensu_check` with name `check-cpu in qa` would be named `check-cpu` and put into the `qa` namespace.
+
+Using composite names is necessary if you wish to have multiple resources with the same name but in different namespaces.
```

Comment:
Add something about if you use composite namespaces and specify `namespace` that `namespace` will take precedence. 

Perhaps also some guidance that if you use namespaces then you should follow this composite pattern. Ideally you wouldn't use namespaces for different environments, you would use hiera data. Is there another use case for the namespace? Don't want to confuse people.

Reply:
I added a note about `namespace` property taking precedence.

I think namespaces can probably be different than puppet environments and unrelated to Hiera. Because roles define namespaces it's easy to imagine giving access controls to different namespaces that would all be defined in one Hiera environment so that the production sensu-backend could manage multiple namespaces for different roles or entities. As for composite vs non-composite, it's eluded to in README that the main benefit to composite is that you can have something like a check have the same name is Sensu for different environments. Besides having the same name in Sensu, there's not real benefit to using composite names if sites decide they want namespaces to be in the name they could just name checks `check-cpu-qa` or something.

Reply:
so for the example should we change `qa` to something like `tenant1` or `namespace_foo`? I don't want to encourage folks to use namespace's for their deployment environments.

Reply:
Ya I can change that to be something unique to a team or geographical location or something not specific to a sort of development workflow.

Reply:
Updated examples in README.